coap-gateway: owner cache integration into service

Danielius1922 opened the pull request.

Danielius1922 committed:
coap-gateway: owner cache integration into service

ondrejtomcik commented on coap-gateway/config.yaml:
```diff
@@ -6,6 +6,7 @@ apis:
     address: "0.0.0.0:5684"
     externalAddress: ""
     maxMessageSize: 262144
+    cacheExpiration: 10m
```
It's not descriptive enough. ownerCachceExpiration? Should it be really limited in the case of CoAP Gateway which should keep the cache open as long as the device is connected? There shouldn't be imho any time-based expiration.

jkralik replied:
It's nice when you connect `1000 devices` at the same time so it will use data from the updated cache.

ondrejtomcik replied:
Decrease to 1m

Danielius1922 replied:
Updated

Danielius1922 committed:
nats: rework configuration

Merge nats/publisher and nats/subscriber configuration
types and create configuration types in nats/client.

Danielius1922 committed:
resource-aggregate/nats: add connection wrapper

Add Client type that wraps *nats.Conn and offers convenience
methods.
Used in coap-gw service to share connection to nats by
publisher and ownerCache.

Danielius1922 committed:
ownerCache: add OwnsDevice method

Danielius1922 committed:
coap-gateway: verify device owner in signIn

Danielius1922 committed:
pkg: add Intersection method to sortedSlice

Danielius1922 committed:
ownerCache: add OwnsDevices method

jkralik commented on resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go:
```diff
@@ -134,45 +135,57 @@ func WithGoPool(goroutinePoolGo eventbus.GoroutinePoolGoFunc) GoroutinePoolGoOpt
+		return nil, fmt.Errorf("cannot create nats connection: %w", err)
+	}
+
+	s, err := NewWithNATS(conn, config.PendingLimits, logger, opts...)
```
who close conn function? `NewWithNATS` I expected that the caller close conn. And with `New` I expected that will be close by subscribers. 
Can we rename `NewWithNATS` to just `New` and drop the old `New`?

Danielius1922 replied:
The current `subscriber.New` is used 14 times and `publisher.New` is used 33 times (mostly in tests though), so when I get rid of the original `New` then some convenience method equivalent would be useful.

Danielius1922 replied:
Solved by creating helper function for tests.

jkralik commented on resource-aggregate/cqrs/eventbus/nats/publisher/publisher.go:
```diff
@@ -53,39 +54,49 @@ func WithMarshaler(dataMarshaler MarshalerFunc) MarshalerOpt {
-func newPublisher(url string, jetstream bool, eventMarshaler MarshalerFunc, options ...nats.Option) (*Publisher, error) {
-	conn, err := nats.Connect(url, options...)
+// Create publisher with existing NATS connection
+func NewWithNATS(conn *nats.Conn, jetstream bool, opts ...Option) (*Publisher, error) {
```
can we rename `NewWithNATS` to `New` and drop the old `New`?

Danielius1922 committed:
resource-aggregate: refactor publisher constructor

Danielius1922 committed:
resource-aggregate: refactor subscriber constructor
